www/include/configuration/configObject/service/formService.php

```diff
@@ -62,36 +62,43 @@ function myDecodeService($arg)
     return html_entity_decode($arg, ENT_QUOTES, "UTF-8");
 }
 
-if (!$centreon->user->admin) {
-    if ($service_id) {
-        $checkres = $pearDB->query("SELECT service_id
-                                        FROM $aclDbName.centreon_acl
-                                        WHERE service_id = " . $pearDB->escape($service_id) . "
-                                        AND group_id IN (" . $acl->getAccessGroupsString() . ")");
-        if (!$checkres->rowCount()) {
-            $msg = new CentreonMsg();
-            $msg->setImage("./img/icons/warning.png");
-            $msg->setTextStyle("bold");
-            $msg->setText(_('You are not allowed to access this service'));
-            return null;
-        }
+if (!$centreon->user->admin && isset($service_id)) {
+    $checkres = $pearDB->query(
+        "SELECT service_id
+        FROM $aclDbName.centreon_acl
+        WHERE service_id = " . $pearDB->escape($service_id) . "
+        AND group_id IN (" . $acl->getAccessGroupsString() . ")"
+    );
```

Comment:
Did this query shouldn't be prepare/bind instead of using query/pearDB->escape ?

Reply:
Yes of course, but this is the original query and modifying it would have involved completely rebuilding the query without going through the $acl->getAccessGroupsString method.